src/main/java/net/starschema/clouddb/jdbc/BQStatement.java

```diff
@@ -193,7 +198,29 @@ private ResultSet executeQueryHelper(String querySql, boolean unlimitedBillingBy
                   .get(projectId, qr.getJobReference().getJobId())
                   .setLocation(qr.getJobReference().getLocation())
                   .execute();
+
+          BiEngineStatistics biEngineStatistics = null;
+          if (referencedJob != null) {
+            JobStatistics2 statistics2 = referencedJob.getStatistics().getQuery();
+            if (statistics2 != null) {
+              biEngineStatistics = statistics2.getBiEngineStatistics();
+            }
+          }
+
+          if (biEngineStatistics != null) {
+            biEngineMode = biEngineStatistics.getBiEngineMode();
+            biEngineReasons = biEngineStatistics.getBiEngineReasons();
+          } else {
```

Comment:
Rather than all these else-clauses, how about initializing them to null above the if statement.

Reply:
If you don't explicitly initialize the members, they will default to null, so you don't need to re-set them to null here (unless there's a way that they could get set and then unset, of course).  Similarly, when declaring `BiEngineStatistics biEngineStatistics = null;` the ` = null` is redundant.

Reply:
It was raising an exception saying it was not set 🤷‍♀️

Reply:
Apologies still brushing up my java skills